crawl-ref/source/describe.cc

```diff
@@ -4011,6 +4011,28 @@ static void _describe_monster_mr(const monster_info& mi, ostringstream &result)
     result << "\n";
 }
 
+/**
+ * Append information about a given monster's habitat to the provided stream.
+ *
+ * @param mi[in]            Player-visible info about the monster in question.
+ * @param result[in,out]    The stringstream to append to.
+ */
+static void _describe_monster_hab(const monster_info& mi, ostringstream &result)
+{
+    habitat_type h = mons_habitat_types(mi.type, mi.base_type);
```

Comment:
super nit: should be `const auto habitat` or `const habitat_type habitat`
 - use const; the lack of const indicates that it's mutated later (but it isn't in this case)
 - `h` is an unclear name; prefer avoiding abbreviations if possible